tests/js/server/dump/dump-test.inc

```diff
@@ -1184,7 +1185,136 @@
           assertTrue(next < 100000000, "expected next uniqid after restore to be less than 100000000, not " + next);
         }
       },
-      
+
+      ////////////////////////////////////////////////////////////////////////////////
+      /// @brief test smart graph restores
+      ////////////////////////////////////////////////////////////////////////////////
+      testEmptySmartGraph: function () {
+        const smartGraphName = "UnitTestsDumpEmptySmartGraph";
+        const g = db._collection('_graphs');
+        console.warn(g.toArray());
+        assertTrue(!!g.exists(smartGraphName));
+
+        const graph = g.document(smartGraphName);
+        assertFalse(graph.isDisjoint);
+        assertFalse(graph.isSatellite);
+        assertTrue(graph.isSmart);
+        assertEqual(graph.replicationFactor, 1);
+        assertEqual(graph.numberOfShards, 5);
+        assertEqual(graph.smartGraphAttribute, 'value');
+        assertEqual(graph.edgeDefinitions.length, 0);
+        assertEqual(graph.orphanCollections.length, 0);
+      },
+
+      testSmartGraphWithoutData: function() {
+        const smartGraphName = "UnitTestDumpSmartGraphWithoutData";
+        const vertexCollName = 'UnitTestDumpSmartVerticesWithoutData';
+        const edgeCollName = 'UnitTestDumpSmartEdgesWithoutData';
+        const orphanCollName = 'UnitTestDumpSmartOrphansWithoutData';
+
+        const g = db._collection('_graphs');
+        assertTrue(!!g.exists(smartGraphName));
+
+        const graph = g.document(smartGraphName);
+        assertFalse(graph.isDisjoint);
+        assertFalse(graph.isSatellite);
+        assertTrue(graph.isSmart);
+        assertEqual(graph.replicationFactor, 1);
+        assertEqual(graph.numberOfShards, 5);
+        assertEqual(graph.smartGraphAttribute, 'value');
```

Comment:
These asserts are the same multiple times, this could be done once in a separate function.

Reply:
Fixed.